src/Repository/ElectedRepresentative/ElectedRepresentativeRepository.php

```diff
@@ -240,46 +247,26 @@ private function withActiveMandatesCondition(QueryBuilder $qb, string $alias = '
         ;
     }
 
-    private function withZoneCondition(QueryBuilder $qb, array $referentTags, string $alias = 'er'): QueryBuilder
+    private function withZoneCondition(QueryBuilder $qb, array $zones, string $alias = 'er'): QueryBuilder
     {
         if (!\in_array('mandate', $qb->getAllAliases(), true)) {
             $qb->leftJoin($alias.'.mandates', 'mandate');
         }
 
-        $hasParis = false;
-        $districtDptCodes = [];
-        foreach ($referentTags as $tag) {
-            if ($districtDptCode = $tag->getDepartmentCodeFromCirconscriptionName()) {
-                $districtDptCodes[] = $districtDptCode;
-            }
-
-            if (0 === mb_strpos($tag->getCode(), '750') || 0 === mb_strpos($tag->getCode(), 'CIRCO_750')) {
-                $hasParis = true;
-
-                break;
-            }
-        }
-
-        $zoneCondition = new Orx();
-        $zoneCondition->add('tag IN (:tags)');
-        $qb->setParameter('tags', $referentTags);
-        // if referent has some Paris tag, we should return elected representatives of all Paris zones
-        if ($hasParis) {
-            $zoneCondition->add('tag.code LIKE :paris_arr OR tag.code LIKE :paris_circo');
-            $qb->setParameter('paris_arr', '750%');
-            $qb->setParameter('paris_circo', 'CIRCO\_750%');
-        }
-
-        if ($districtDptCodes) {
-            $zoneCondition->add('tag.code IN (:districtDptCodes)');
-            $qb->setParameter('districtDptCodes', $districtDptCodes);
+        if ($zones) {
+            $qb
+                ->innerJoin('mandate.geoZone', 'geo_zone')
+                ->leftJoin('geo_zone.children', 'geo_zone_child')
+                ->andWhere(
+                    $qb->expr()->orX(
+                        $qb->expr()->in('geo_zone.id', ':zones'),
+                        $qb->expr()->in('geo_zone_child.id', ':zones'),
```

Comment:
We need to check if parent zones are in the managed zoned and not children.
For exemple, if managed zone is department we should find all mandates in city, EPCI and department zone.

Reply:
thanks, fixed in 15eba7ac37d2636b56e493269e0ec3ae10fcf54a